assets/csidriveroperators/aws-ebs/hypershift/guest/kustomization.yaml

```diff
@@ -10,12 +10,10 @@ patches:
     target:
       kind: Role
       version: v1
-  - patch: |-
+  - patch: |
       $patch: delete
-      apiVersion: apps/v1
-      kind: Deployment
+      kind: Kustomization
       metadata:
```

Comment:
remove entire `Kustomization` altogether? This reads weird. I haven't tried this..

Reply:
I agree that it's weird, however, we currently do this to indicate that assets marked up with `storage.openshift.io/remove-from: mgmt` annotation should be dropped/not included in the management cluster in a HCP cluster.

https://github.com/openshift/cluster-storage-operator/blob/0ccc4bc8f4cc299ab8db8294c0234fa8a7451284/assets/csidriveroperators/aws-ebs/hypershift/mgmt/kustomization.yaml#L23-L29

https://github.com/openshift/cluster-storage-operator/blob/0ccc4bc8f4cc299ab8db8294c0234fa8a7451284/assets/csidriveroperators/aws-ebs/base/10_cr.yaml#L6

I'm just copying this pattern and using it for guest cluster assets. It seems better to at least be consistent: I didn't see any reason why we should use `storage.openshift.io/remove-from: mgmt` for management cluster assets but not `storage.openshift.io/remove-from: guest` for guest cluster assets. As usual with me, there's more context in my commit messages:

https://github.com/openshift/cluster-storage-operator/pull/526/commits/f901a4a5532ce73eb9d7b01d05a8ff44e04c2e2b

(I need to start copy-pasting more context from commit messages into PR message :see_no_evil:)

Reply:
This is one the codes, I wrote and forgot about it and now I had trouble understanding it. I will add some comment as a follow up.